### DISCUSSION TITLE
How do pass user id to reference table in POST METHOD

### DISCUSSION THREAD

**Discussion by groovyrb** [directus/directus#4546] [Archive]
I am creating a new user from API and At the same time, I need to create data of the user to another table you can find my JSON below! here $new_user_created_id I need to pass which is created recently so How can I get or pass the last inserted id in API so my mapping or relation does not break? Need #help or how to achieve this?

```json
{
    "status": "active",
    "first_name": "keep",
    "last_name": "Smiling",
    "email": "keepsmiling@gmail.com",
    "password": "jhdjshfjsfbsjb",
    "role": "3",
    "title": "Dreemnj",
    "user_data": {
        "phone_number": "2566522560",
        "users": **$new_user_created_id**
    }
}
```

**Comment by rijkvanzanten** [comment]
There's no way to do that currently. You could however go "the other way" by inserting into the `user_data` table, and do the following:

```json
POST /items/user_data

{
  "phone_number": "2566522560",
  "user": {
      "status": "active",
      "first_name": "keep",
      "last_name": "Smiling",
      "email": "keepsmiling@gmail.com",
      "password": "jhdjshfjsfbsjb",
      "role": "3",
      "title": "Dreemnj"
  }
}
```

(I'm making some assumptions about your schema here, not sure if `user` exists / is an m2o, not sure if `user_data` is the name of the collection)

**Reply by groovyrb** [reply]
Hello Thanks for the reply
user_data is a mapping field(M2O) that I have created in the directus_users **table** which is mapped to user_meta **table** 
In user_meta I kept other details of the user and which is mapped to another table or collection. So in this way, I make 3-4 table  which are connecteded in a flow

Work Flow:  user **->** user_meta **->** Branch,Products (**collection**)

**Reply by rijkvanzanten** [reply]
Out of curiosity, if the user_meta is always connected to a single user, why not "just" add those fields to directus_users directly? Saves you some trouble

**Comment by OzzyTheGiant** [comment]
Yeah this needs fixing, not just being able to post related data, but just being able to get a response body back from POST methods. I was  trying to rely on chained web requests but the response body on a POST request doesn't return the newly created record with it's id, so I can't even get the needed foreign key id for other records, and that's because the docs mention that a body is to be returned

**Reply by rijkvanzanten** [reply]
This is already possible.
Make sure your user has _read_ permissions as well as write permissions if you want the created item to be _read_ after create 👍🏻

**Reply by OzzyTheGiant** [reply]
Ah that makes more sense. I would assume that such data would be returned back regardless since it's data you just created but if that's the case, I would think maybe the docs would need to specify what data you get back based on permissions

**Reply by rijkvanzanten** [reply]
> since it's data you just created

Well, it's not! You can have all sorts of hooks, automated fields (like generated UUIDs), relational data, etc etc that isn't actually provided by the user
